Update getrandom to 0.2

josephlr opened the pull request:
Per discussion in #889, we are keeping getrandom optional. Users can now specify custom RNG handlers for `getrandom` on any target. However, not all users will want `OsRng`, so `getrandom` needs to be optional.

Signed-off-by: Joe Richey <joerichey@google.com>

dhardy commented:
Thanks @josephlr.

The codes `CUSTOM_START` and `INTERNAL_START` can now be re-exported from `getrandom` I believe. But, see #984: we should probably document exactly what these values are and assert that they will never change.

vks commented:
@dhardy I don't think we can re-export them as long as `getrandom` is an optional dependency. Instead, I documented the error codes as being identical to the ones in `getrandom` and added a test to make sure.

josephlr commented:
> @dhardy I don't think we can re-export them as long as `getrandom` is an optional dependency. Instead, I documented the error codes as being identical to the ones in `getrandom` and added a test to make sure.

Thanks! I also updated the comments to link to the getrandom documentation.

vks commented on CHANGELOG.md:
```diff
@@ -16,6 +16,7 @@ You may also find the [Upgrade Guide](https://rust-random.github.io/book/update.
 - Implement weighted sampling without replacement (#976, #1013)
 
 ### Changes
+- `getrandom` updated to v0.2
```
This could mention this PR by issue number.

josephlr replied:
Done
